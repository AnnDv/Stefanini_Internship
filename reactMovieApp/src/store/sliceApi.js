import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { POPULAR, BASE_URL, SEARCH_URL } from '../utils';

export const movieApi = createApi({
  reducerPath: 'movieApi',
  baseQuery: fetchBaseQuery({ baseUrl: BASE_URL }),
  endpoints: (builder) => ({
    getPopularMovies: builder.query({
      query: () => POPULAR,
    }),
    getSearchMovie: builder.query({
      query: (query) => ({
        url: SEARCH_URL,
        params: { query },
      }),
    }),
  }),
});

export const { useGetPopularMoviesQuery, useLazyGetSearchMovieQuery } = movieApi;
