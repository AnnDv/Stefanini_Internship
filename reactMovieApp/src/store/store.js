import { configureStore } from '@reduxjs/toolkit';
// import apiReducer from './features/apiSlice';
import { setupListeners } from '@reduxjs/toolkit/dist/query';
import { movieApi } from './sliceApi';

export const middlewares = [movieApi.reducerPath];

const store = configureStore({
  reducer: {
    [movieApi.reducerPath]: movieApi.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(movieApi.middleware),
  // reducer: {
  //   movie: apiReducer,
  // },
});

setupListeners(store.dispatch);

export default store;
