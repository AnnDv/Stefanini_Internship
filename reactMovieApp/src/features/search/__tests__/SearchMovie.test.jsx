import React from 'react';
import fetchMock from 'jest-fetch-mock';
import { Provider } from 'react-redux';
import { fireEvent, render } from '@testing-library/react';
import SearchBox from '../SearchMovie.jsx';
import store from '../../../store/store';

describe('SearchBox', () => {
  beforeEach(() => {
    fetchMock.resetMocks();
  });
  const correctInput = 'star';

  it('should find a movie', () => {
    const mockOnChange = jest.fn();
    const { getByTestId, baseElement } = render(
        <Provider store={store}>
          <SearchBox setSearch={mockOnChange}/>
        </Provider>,
    );

    const input = getByTestId('search');

    fireEvent.change(input, { target: { value: 'star' } });

    expect(mockOnChange).toBeCalledWith(correctInput);
    expect(baseElement).toMatchSnapshot();
  });
});
