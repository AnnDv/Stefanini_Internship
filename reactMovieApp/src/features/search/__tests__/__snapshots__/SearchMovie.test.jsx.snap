// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should find a movie 1`] = `
<body>
  <div>
    <div
      class="form"
    >
      <input
        class="search_form"
        data-testid="search"
        placeholder="Type to search..."
        type="search"
        value=""
      />
    </div>
  </div>
</body>
`;
